conan/tools/cmake/cmakedeps/templates/target_configuration.py

```diff
@@ -21,12 +21,17 @@ def filename(self):
     def context(self):
         deps_targets_names = self.get_deps_targets_names() \
             if not self.conanfile.is_build_context else []
+
+        components_targets_names = self.get_declared_components_targets_names()
+        components_variables_names = [components_variables_name.replace("::", "_") for
+                                      components_variables_name in components_targets_names]
+        components_names = list(zip(components_targets_names, components_variables_names))
```

Comment:
This is the same, isn't it?

```suggestion
        components_names = [(components_variables_name, components_variables_name.replace("::", "_"))
                             for components_variables_name in components_targets_names]
```